src/essreflectometry/types.py

```diff
@@ -0,0 +1,103 @@
+from typing import NewType, TypeVar
+
+import sciline
+import scipp as sc
+
+Reference = NewType('Reference', str)
+Sample = NewType('Sample', str)
+Run = TypeVar('Run', Reference, Sample)
+
+
+class RawData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event time data from nexus file,
+    binned by `detector_number` (pixel of the detector frame)."""
+
+
+class WavelengthData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelengths computed for every event,
+    binned by `detector_number` (pixel of the detector frame)"""
+
+
+class ThetaData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelengths and scattering angle computed for each event,
+    binned by `detector_number` (pixel of the detector frame)"""
+
+
+class QData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelength, incidence angle and momentum transfer computed per event.
+    Binned by momentum transfer according to the QBins provider,
+    and by `detector_number` (pixel of the detector frame).
+    """
+
+
+class FootprintCorrected(sciline.Scope[Run, sc.DataArray], sc.DataArray):
```

Comment:
```suggestion
class FootprintCorrectedData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
```
maybe? Otherwise one might think it is a boolean indicating whether the correction was applied?